returnn/tf/util/data.py

```diff
@@ -3471,6 +3471,7 @@ def copy_add_dim_by_tag(self, dim_tag, unbroadcast=False, axis=None):
     if dim_tag.is_batch_dim():
       if unbroadcast:
         batch_info = dim_tag.src_data.batch if dim_tag.src_data else None
+        batch_info = batch_info or dim_tag.batch
         return self.copy_add_batch_dim(batch_dim_axis=axis, batch=batch_info, dim_tag=dim_tag)
```

Comment:
This looks fine in general. I would even go further and say that `dim_tag.batch` should be preferred when it is set. Maybe we even don't need the other code at all anymore.
```suggestion
        return self.copy_add_batch_dim(batch_dim_axis=axis, batch=dim_tag.batch, dim_tag=dim_tag)
```